### DISCUSSION TITLE
blank page after login with wrong time

### DISCUSSION THREAD

**Discussion by gonboy** [cockpit-project/cockpit#18060] [Q&A] [answered]
### Explain what happens

the console infomation:
```bash
cockpit.js:348 WebSocket connection to 'wss://47.240.80.118:9090/cockpit/socket' failed:
received unsupported version in init message: undefined
transport closed, dropped message:  
{"bus":"internal","payload":"dbus-json3","command":"open","channel":"1","flow-control":true}
received message before init:  hint
Error reading machine id 
{
    "problem": "disconnected",
    "message": "Server has closed the connection."
}
get NTPSynchronized property
{
    "problem": "disconnected",
    "name": null,
    "message": "Server has closed the connection."
}
```

### Where is the problem in Cockpit?

None

### Server operating system

None

**Comment by gonboy** [comment]
I update the system time by NTP.

that's OK.

why ???

**Reply by martinpitt** [reply]
For future reference, which time was wrong? The one on your server, or on the client that runs your web browser? What browser do you use?

I find it really strange that the browser accept an invalid certificate for the html/css pages, but not for the websocket. Something is off here.

Thanks!

**Comment by garrett** [comment, marked as answer]
If your system time was wildly off, all sorts of issues could happen. It's not a Cockpit-specific bug if that was the case.

It's a common enough bug that there are a few xkcd comic about it:

![xkcd comic about code crashing when the time is wildly wrong](https://user-images.githubusercontent.com/10246/205860436-c2e24976-b3f9-4477-a6a8-74d1f1f0e8d4.png)
https://m.xkcd.com/376/

![xkcd comic about dates being wrong misinterpreted](https://user-images.githubusercontent.com/10246/205861190-3a1cbb79-f867-40dc-8190-d85f6c1f40ee.png)
https://m.xkcd.com/2676/

(This second one is because when the time is not set properly, files show up as from 

These two things (incorrect dates and weird effects as the result of very wrong dates) are some of the big reasons why there's NTP to sync up times. (It's also to have extremely accurate times.)

If you have this problem when time is set correctly, please [check the FAQ](https://cockpit-project.org/faq.html#login) and reopen the issue or file a new one if it's not related to an incorrect system configuration.
